ibc-core/ics04-channel/types/src/error.rs

```diff
@@ -3,173 +3,132 @@
 use displaydoc::Display;
 use ibc_core_client_types::error::ClientError;
 use ibc_core_client_types::Height;
-use ibc_core_connection_types::error as connection_error;
 use ibc_core_host_types::error::IdentifierError;
-use ibc_core_host_types::identifiers::{ChannelId, ConnectionId, PortId, Sequence};
+use ibc_core_host_types::identifiers::{ChannelId, PortId, Sequence};
 use ibc_primitives::prelude::*;
 use ibc_primitives::{Timestamp, TimestampError};
 
 use super::channel::Counterparty;
 use super::timeout::TimeoutHeight;
-use crate::channel::State;
+use crate::commitment::PacketCommitment;
 use crate::timeout::TimeoutTimestamp;
 use crate::Version;
 
 #[derive(Debug, Display)]
 pub enum ChannelError {
-    /// invalid channel end: `{channel_end}`
-    InvalidChannelEnd { channel_end: String },
+    /// application module error: `{description}`
+    AppModule { description: String },
```

Comment:
This variant exists solely to convert a `TokenTransferError` into a `ContextError`: `ChannelError` merely serves as an intermediary for the conversion. Is this a pattern that makes sense, or should we do away with it?

Reply:
Right. It serves the purpose you described, but it's causing the `TokenTransferError` and other app-level errors to be converted into strings. This isn't ideal because it undermines the existing `Error` enum definitions at the application level. As highlighted in the critical review, a next step should be to explore ways to reduce the number of nested enums and move away from using these universal variants.